distributed/cli/dask_scheduler.py

```diff
@@ -70,11 +72,14 @@ def del_pid_file():
         with ignoring(ImportError):
             from distributed.bokeh.scheduler import BokehScheduler
             services[('bokeh', bokeh_internal_port)] = BokehScheduler
-    scheduler = Scheduler(loop=loop, services=services)
+    scheduler = Scheduler(loop=loop, services=services,
+                          scheduler_file=scheduler_file)
     scheduler.start(addr)
 
     bokeh_proc = None
     if _bokeh:
+        if bokeh_port == 0:
+            bokeh_port = open_port()
```

Comment:
AFAIU, our user's usecase is for programmatic starting of a scheduler, so something that works most of the time isn't really satisfactory. Two solutions I can think of:
1. have Bokeh grow something like the scheduler file argument
2. allow Bokeh to take an existing (bound) socket

The first solution sounds less hairy than the second.